rsync_to_rdisc.py

```diff
@@ -69,124 +63,137 @@ def send_email(sender, receivers, subject, text, attachment=None):
-            output = settings.folder_dic[item],
-            bgarray=settings.bgarray,
-            log = log,
-            errorlog = errorlog,
-            temperror = temperror
-            ) 
-        state = sync(action1, action2, folder, processed, item)
-
-os.system("rm {}".format(running))
+        print("Mount is lost.")
+        make_mail("mount", ["lost"])
+        sys.exit()
+
+    """ If daemon is running exit, else create transfer.running file and continue """
+    wkdir = settings.wkdir
+    run_file = "{}/transfer.running".format(wkdir)
+    if os.path.isfile(run_file):
+        sys.exit()
+    else:
+        os.system("touch {}".format(run_file))
+
+    """ Make dictionairy of transferred_runs.txt file, or create transferred_runs.txt if not present """
+    transferred_dic = {}
+    if os.path.isfile(str(wkdir) + "transferred_runs.txt"):
+        runs = open(str(wkdir) + "transferred_runs.txt", "r").readlines()
+        for line in runs:
+            transferred_dic[line.rstrip()] = ""
+    else:
+        new_file = open(str(wkdir) + "transferred_runs.txt", "w")
+        new_file.close()
+
+    """ Get folders to be transfer from HPC """
+    client = SSHClient()
+    client.load_host_keys(settings.host_keys)
+    client.load_system_host_keys()
+    client.connect(settings.server, username=settings.user)
+
+    to_be_transferred = {}
+    folder_dic = settings.folder_dic
+    for folder in folder_dic:
+        stdin, stdout, stderr = client.exec_command("ls {}".format(folder_dic[folder][0]))
+        folders = stdout.read().decode("utf8")
+        for item in folders.split():
+            combined = "{0}_{1}".format(item.split()[-1], folder)
+            if combined not in transferred_dic:
+                to_be_transferred[item.split()[-1]] = folder
+
+    for item in to_be_transferred:
+        print("{0} from {1} will be transfer to LABarray".format(item, to_be_transferred[item]))
+
+    """Rsync folders from HPC to bgarray"""
+    log = settings.log
+    errorlog = settings.errorlog
+    temperror = settings.temperror
+    date = str(datetime.datetime.now()).split(".")[0]
+    bgarray_log_file = "{bgarray}/{log}".format(bgarray=settings.bgarray, log=log)
+    remove_run_file = True
+
+    for run in to_be_transferred:
+        continue_rsync = True
+        state = [""]
+        folder = settings.folder_dic[to_be_transferred[run]]
+
+        action = ("rsync -rahuL --stats {user}@{server}:{path}{run} {output}/ \
+            1>> {bgarray}/{log} 2>> {bgarray}/{errorlog} 2> {temperror}".format(
+                user=settings.user,
+                server=settings.server,
+                path=folder[0],
+                run=run,
+                output=folder[1],
+                bgarray=settings.bgarray,
+                log=log,
+                errorlog=errorlog,
+                temperror=temperror
+            )
+        )
+
+        if folder[2]:  # check if certain files need to be present in folder:
```

Comment:
Nice feature: Make it possible to check on multiple files?

Reply:
Included